From fe03ab4c14e71cb08210159a943a6edded6cdc4d Mon Sep 17 00:00:00 2001
From: Werner Koch <wk@gnupg.org>
Date: Mon, 21 Dec 2020 17:54:06 +0100
Subject: [PATCH] Fix a possible segv in case of an unknown CMS object.

* src/cms.c (ksba_cms_get_enc_val): Fix strcmp.
--

Fixes-commit: 401dc58d3d55ed58a0ac4e1f286a7e19ed9e956c
---
 src/cms.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cms.c b/src/cms.c
index 6a563bf..7f63311 100644
--- a/src/cms.c
+++ b/src/cms.c
@@ -1482,7 +1482,7 @@ ksba_cms_get_enc_val (ksba_cms_t cms, int idx)
       /* gpgrt_log_debug ("%s: encryptedKey:\n", __func__); */
       /* dbg_print_sexp (string); */
     }
-  else if (!strcmp (n->name, "kekri"))
+  else if (!strcmp (root->name, "kekri"))
     return NULL; /*GPG_ERR_UNSUPPORTED_CMS_OBJ*/
   else
     return NULL; /*GPG_ERR_INV_CMS_OBJ*/
-- 
2.29.2

